packages/flutter_test/lib/src/test_pointer.dart

```diff
@@ -36,24 +67,113 @@ class TestPointer {
     );
   }
 
-  PointerUpEvent up({ Duration timeStamp: const Duration() }) {
+  /// Create a [PointerUpEvent].
+  ///
+  /// By default, the time stamp on the event is [Duration.ZERO]. You
+  /// can give a specific time stamp by passing the `timeStamp`
+  /// argument.
+  ///
+  /// The object is no longer usable after this method has been called.
+  PointerUpEvent up({ Duration timeStamp: Duration.ZERO }) {
     assert(isDown);
-    isDown = false;
+    _isDown = false;
     return new PointerUpEvent(
       timeStamp: timeStamp,
       pointer: pointer,
       position: location
     );
   }
 
-  PointerCancelEvent cancel({ Duration timeStamp: const Duration() }) {
+  /// Create a [PointerCancelEvent].
+  ///
+  /// By default, the time stamp on the event is [Duration.ZERO]. You
+  /// can give a specific time stamp by passing the `timeStamp`
+  /// argument.
+  ///
+  /// The object is no longer usable after this method has been called.
+  PointerCancelEvent cancel({ Duration timeStamp: Duration.ZERO }) {
     assert(isDown);
-    isDown = false;
+    _isDown = false;
     return new PointerCancelEvent(
       timeStamp: timeStamp,
       pointer: pointer,
       position: location
     );
   }
+}
+
+/// A class for performing gestures in tests.
+///
+/// The simplest way to create a [TestGesture] is to call
+/// [WidgetTester.startGesture].
+class TestGesture {
+  /// Create a [TestGesture] by starting with a pointerDown at the
+  /// given point.
+  ///
+  /// By default, the pointer ID used is 1. This can be overridden by
+  /// providing the `pointer` argument.
```

Comment:
That would cause this to hyperlink to a field or method called 'pointer' if we ever added one.
FWIW, I'm thinking of lobbying for a hint that warns if dartdocs use `[]` but fail to hyperlink.